graph/telemetry/istio/appender/service_entry.go

```diff
@@ -72,14 +72,46 @@ func (a ServiceEntryAppender) getServiceEntry(service string, globalInfo *graph.
 						location = "MESH_INTERNAL"
 					}
 					for _, host := range entry.Spec.Hosts.([]interface{}) {
-						serviceEntries[host.(string)] = location
+						serviceEntryHosts = append(serviceEntryHosts, serviceEntryHost{
+							location: location,
+							host:     host.(string),
+						})
 					}
 				}
 			}
 		}
-		log.Tracef("Found [%v] service entries", len(serviceEntries))
+		globalInfo.Vendor[serviceEntryHostsKey] = serviceEntryHosts
 	}
 
-	location, ok := serviceEntries[service]
-	return location, ok
+	for _, serviceEntryHost := range serviceEntryHosts {
+		// handle exact match
+		if serviceEntryHost.host == serviceName {
+			return serviceEntryHost.location, true
+		}
+		// handle serviceName prefix (e.g. host = serviceName.namespace.svc.cluster.local)
+		if serviceEntryHost.location == "MESH_INTERNAL" {
+			if strings.Split(serviceEntryHost.host, ".")[0] == serviceName {
+				return serviceEntryHost.location, true
+			}
+		}
+		// handle wildcard
+		if serviceEntryHost.location == "MESH_EXTERNAL" {
+			hostTokens := strings.Split(serviceEntryHost.host, ".")
+			serviceNameTokens := strings.Split(serviceName, ".")
+			match := len(hostTokens) == len(serviceNameTokens)
```

Comment:
Can the hostTokens be shorter than the serviceName?
e.g.
Would hostToken= ".my-domain.com" match serviceName="one.two.three.my-domain.com" ?
What about the other side, "mail." match serviceName="mail.google.com" ?

Reply:
I actually don't know.  Perhaps a different approach is needed here.  It may be that the wildcard is allowed only at the beginning at that it should be able to match anything as long as the domain "tail" matches exactly.  Let me know what you think after you read up.

Reply:
Sounds good!

Reply:
@josejulio After reading it again, and also [reading this](https://istio.io/docs/examples/advanced-gateways/wildcard-egress-hosts/), I think I should change the approach to just be a prefix wildcard that matches any prefix in front of the common domain "tail".   I'll make the change...

Reply:
@josejulio I updated the wildcard logic to just handle prefix wildcarding. Please take a look.